sql/modules/Roles.sql

```diff
@@ -1280,10 +1289,11 @@ IF TG_OP = 'DELETE' THEN
 ELSE
    -- super user and database owner (group members)
    -- don't need access enforcement
-   IF pg_has_role((select rolname
-                     from pg_database db inner join pg_roles rol
-                       on db.datdba = rol.oid
-                    where db.datname = current_database()),
+   IF pg_has_role((SELECT rolname
+                     FROM pg_database db
+               INNER JOIN pg_roles rol
+                       ON db.datdba = rol.oid
+                    WHERE db.datname = current_database()),
```

Comment:
By the same reasoning, should we not cache the value of the `current_database` function?

Reply:
This code is seldom used and the optimizer seems to behave correctly for its internal STABLE functions. Splitting the code in 2 statements actually makes it longer.

Reply:
Ok. Then this is a non-issue.